### PR TITLE
Add options calendarFocus and preventUnnecessaryRefocus

### PR DESCRIPTION
**Pull request by lexispike**
## Types of changes

What types of changes does your code introduce?

_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Description
Add options to 
- allow specified calendar focus directions ('forwards' or 'backwards') and 
- allow prevention of unnecessary refocus on selection of a date in viewing range (for non-scrolled calendars)

> Related Issues: #427 & #268

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -127,6 +127,8 @@ preview(DateRange)                   | Object    |                  | displays a
 showPreview(DateRange)               | bool      | true             | visibility of preview
 editableDateInputs(Calendar)         | bool      | false            | whether dates can be edited in the Calendar's input fields
 dragSelectionEnabled(Calendar)       | bool      | true             | whether dates can be selected via drag n drop
+calendarFocus(Calendar)              | String    | 'forwards'       | Whether calendar focus month should be forward-driven or backwards-driven. can be 'forwards' or 'backwards'
+preventUnnecessaryRefocus(Calendar)  | bool      | false            | prevents unneceessary refocus of shown range on selection
```
Comment (kamyar) [inline README.md:131]: `Unnecessary` int he prop name seems redundant and I think we can rename the prop to exclude that so it is more concise.
wdyt?

**Comment by kamyar** [comment]
Thanks for the fix, I just resolved some conflict and one small comment. :)

File: README.md (hunk 1 above)
Comment (lexispike) [reply README.md:131]: Thanks for the feedback! How about something more like `preventSnapRefocus`?

Comment (kamyar) [reply README.md:131]: sure, that looks better :)

File: src/components/Calendar/index.js [hunk 2]
```diff
@@ -77,6 +78,18 @@ class Calendar extends PureComponent {
   }
   focusToDate = (date, props = this.props, preventUnnecessary = true) => {
     if (!props.scroll.enabled) {
+      if (preventUnnecessary && props.preventUnnecessaryRefocus) {
+        const focusedDateDiff = differenceInCalendarMonths(
+          date,
+          this.state.focusedDate,
+          this.dateOptions
+        );
```
Comment (kamyar) [inline src/components/Calendar/index.js:86]: `differenceInCalendarMonths` seems to take only two arguments and options is redundant, can you remove that? :) 
(dateLeft: Date | number, dateRight: Date | number)

File: src/components/Calendar/index.js [hunk 3]
```diff
@@ -484,7 +497,12 @@ class Calendar extends PureComponent {
               isVertical ? this.styles.monthsVertical : this.styles.monthsHorizontal
             )}>
             {new Array(this.props.months).fill(null).map((_, i) => {
-              const monthStep = addMonths(this.state.focusedDate, i);
+              let monthStep;
+              if (this.props.calendarFocus === 'forwards') {
+                monthStep = addMonths(this.state.focusedDate, i);
+              } else {
+                monthStep = subMonths(this.state.focusedDate, this.props.months - 1 - i);
+              }
```
Comment (kamyar) [inline src/components/Calendar/index.js:505]: let's only have an `if` statement to override the value if calendarFocus is `backwards`.
More granule change would help us avoid possible bugs imho.

```suggestion
              let monthStep = addMonths(this.state.focusedDate, i);;
              if (this.props.calendarFocus === 'backwards') {
                monthStep = subMonths(this.state.focusedDate, this.props.months - 1 - i);
              }
```

**Comment by kamyar** [comment]
Can you kindly add two small gifs demoing the old and new way this behaves also to the PR description?
Please let me know if you need any help

**Comment by AnderUstarroz** [comment]
Will this PR be merged anytime soon? If the new **preventUnnecessaryRefocus** flag prevents the "annoying" effect of changing the month when selecting the start date ([ongoing issue 427](https://github.com/hypeserver/react-date-range/issues/427)), this will bring joy to many people.

**Comment by lexispike** [comment]
https://jmp.sh/2leh8i8 GIF of example calendar with new options :)

**Comment by lexispike** [comment]
Sorry for the delay on this, should be good to go!

**Comment by lexispike** [comment]
Also, a before (or with `preventSnapRefocus` disabled): https://jmp.sh/dkpptq0
after (with it enabled, a longer gif): https://jmp.sh/VoZMDET

**Review by kamyar — APPROVED** [review]

**Comment by kamyar** [comment]
Thank you for the contribution and addressing the comments! 🎉 
I will try to cut a release today 🤞 :octocat:

**Comment by kamyar** [comment]
(Also sorry for the delay in coming back to this, has been a busy 1-2 months 😓 , thank you for the patience)

**Comment by lucasNeuhauss** [comment]
Hi.
Will there be a new npm version release applying these changes soon? Looking forward to it.
Thanks.

**Comment by kamyar** [comment]
Hey, sorry did not get a chance to do it last weekend.
I plan to do this weekend, but if anyone would like to get involved, these are the changes needed: https://github.com/hypeserver/react-date-range/pull/497/files

- bump to 1.4.0
- update changelog file
Feel free to open a PR and I can review, approve, merge & trigger push

**Comment by kamyar** [comment]
Hey all,
v1.4.0 is released ✨, please take a look and let us know if you see something out of place. 
Have a nice weekend!

**Comment by AnderUstarroz** [comment]
Working great! Thanks @lexispike, thanks @kamyar.

**Comment by AnderUstarroz** [comment]
This is still an ongoing issue @kamyar.  Please have a look at the  the following scenario :

- On mobile, displaying a single month.
- Click the start date as the last day of the month.
- Click the end date as the first day of the next month.


**Expected outcome**
Stays on the next month.

**Actual outcome:**
Jumps to the first month (even thought I am using **alendarFocus={"backwards"}** and **preventSnapRefocus={true}**)

**Pull request merged**